sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -353,6 +353,13 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val PARQUET_FILTER_PUSHDOWN_DATE_ENABLED = buildConf("spark.sql.parquet.filterPushdown.date")
+    .doc("If true, enables Parquet filter push-down optimization for Date. " +
+      "This configuration only has an effect when 'spark.sql.parquet.filterPushdown' is enabled.")
+    .internal()
+    .booleanConf
+    .createWithDefault(false)
```

Comment:
an internal by-default-false conf usually means it's not available for users...

Reply:
@dongjoon-hyun @HyukjinKwon suggested such configuration could be better to start with "false", but actually ORC does support Date push down by default, any idea?

Reply:
I am fine either way.

Reply:
@yucai . The reason is that `spark.sql.orc.filterPushdown` is still `false` in Spark 2.3 while `spark.sql.parquet.filterPushdown` is `true`. We don't know this is safe or not.

Anyway, we have 6 or more months for Apache Spark 2.4. We may enable this in `master` branch temporarily for testing purpose only, and are able to disable this at the last moment of 2.4 release like we did about ORC conf if there is some issue.

BTW, did you use this in your company a lot in production?

Reply:
I think it's common that we turn on new feature by default, if there is no known regression. And turn it off if we find regression later.

Reply:
If you think so, +1. :)

BTW, based on Apache Spark way, I assume that this will not land on `branch-2.3` with `spark.sql.parquet.filterPushdown.date=true`.

Reply:
This is not a bug fix. We will not backport it to branch-2.3

Reply:
Great! Thank you for confirmation.

Reply:
@dongjoon-hyun, we are investigating to use this feature in some kind of eBay's queries, if its performance is good, will benefit a lot.
As per our discussion, I will turn it on by default, thanks very much!